packages/react-native-reanimated/android/build.gradle

```diff
@@ -78,29 +33,21 @@ def resolveReactNativeDirectory() {
 
     // Fallback to node resolver for custom directory structures like monorepos.
     def reactNativePackage = file(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim())
-    if(reactNativePackage.exists()) {
+    if (reactNativePackage.exists()) {
         return reactNativePackage.parentFile
     }
 
     throw new GradleException(
-        "[Reanimated] Unable to resolve react-native location in node_modules. You should project extension property (in `app/build.gradle`) `REACT_NATIVE_NODE_MODULES_DIR` with path to react-native."
+        "[Reanimated] Unable to resolve react-native location in node_modules. You should set project extension property (in `app/build.gradle`) named `REACT_NATIVE_NODE_MODULES_DIR` with the path to react-native in node_modules."
     )
 }
 
-def getPlaygroundAppName() { // only for the development
-    String playgroundAppName = ""
-    try {
-        rootProject.getSubprojects().forEach({project ->
-            if (project.plugins.hasPlugin("com.android.application")) {
-                var projectCatalogAbsolutePath = project.projectDir.toString().replace("/android/app", "")
-                var slashPosition = projectCatalogAbsolutePath.lastIndexOf("/")
-                playgroundAppName = projectCatalogAbsolutePath.substring(slashPosition + 1)
-            }
-        })
-    } catch (_) {
-        throw new GradleException("[Reanimated] Couldn't determine playground app name.")
-    }
-    return playgroundAppName
+def getReactNativeMinorVersion() {
+    def reactNativeRootDir = resolveReactNativeDirectory()
+    def reactProperties = new Properties()
+    file("$reactNativeRootDir/ReactAndroid/gradle.properties").withInputStream { reactProperties.load(it) }
+    def reactNativeVersion = reactProperties.getProperty("VERSION_NAME")
+    return reactNativeVersion.startsWith("0.0.0-") ? 1000 : reactNativeVersion.split("\\.")[1].toInteger()
 }
```

Comment:
Moved React Native version detection logic from top-level scope to `getReactNativeMinorVersion` function